src/markdownItPlugin.js

```diff
@@ -0,0 +1,137 @@
+// Inspired by https://github.com/ulivz/vuepress-plugin-flowchart
+
+const { hash } = require('@vuepress/shared-utils')
+
+module.exports = function mermaidjsPlugin (md, options = {}) {
+
+  // Handle ```mermaid blocks
+  const fence = md.renderer.rules.fence
+  md.renderer.rules.fence = (...args) => {
+      const [tokens, idx] = args
+      const { info } = tokens[idx]
+      if (info.trim(' ') === 'mermaid') {
+        return render(tokens, idx)
+      }
+      const rawCode = fence(...args)
+      return `${rawCode}`
+  }
+
+  // Handle <mermaid> blocks
+  options = options || {}
+  const openMarker = options.openMarker || '<mermaid'  // Not ending with '>' to allow attributes
+  const openChar = openMarker.charCodeAt(0)
+  const closeMarker = options.closeMarker || '</mermaid>'
+  const closeChar = closeMarker.charCodeAt(0)
+
+  function render (tokens, idx, options, env, self) {
+    const token = tokens[idx]
+    const key = `mermaid_${hash(idx)}`
+    const { content } = token
+    md.$dataBlock[key] = content
+    return `<Mermaid id="${key}" :graph="$dataBlock.${key}"></Mermaid>`
+  }
+
+  function uml (state, startLine, endLine, silent) {
```

Comment:
Same here, doc comment please. Also, I would prefer having this renamed to mermaidRender or something to be a little more explicit.

Reply:
Good point. I've added comments. 😄